### ISSUE TITLE
Add an ability to generate DataFetcherResult for a specific fields in Java

### ISSUE THREAD

**Issue by kirill071992** [kobylynskyi/graphql-java-codegen#1416]
**Is your feature request related to a problem? Please describe.**
We have some cases when local context is required for Data fetching, to be able to pass some specific data for child resolvers. In this case, global context from DataFetchingEnvironment is not an option.

An example of using a local context:
```java
return DataFetcherResult.<MobileLandingBlockGraph>newResult()
                .data(new BooksMobileLandingBlockGQL(groups.stream()
                        .map((Function<BooksProductGroup, BooksProductGroupGraph>) BooksProductGroupGQL::new)
                        .toList()))
                .localContext(booksProductService.populateContextForMobile(input.getInput(), "advantures"))
                .build();
``` 

As DataFetcherResult is not supported in the current version of the generator, we have to use the following to be able to generate it for a specific field in GraphQl schema:

```java
abstract class GraphQLCodegenFixesGradleTask : DefaultTask() {
    @get:InputFile
    abstract val inputFile: RegularFileProperty

    @TaskAction
    fun updateMobileLandingGridDataGraph() {
        val findStr = "java.util.List<MobileLandingBlockGraph>" // Specify the string to find
        val replaceStr = "java.util.List<graphql.execution.DataFetcherResult<MobileLandingBlockGraph>>" // Specify the string to replace with

        val file = inputFile.get().asFile
        val content = file.readText()
        val updatedContent = content.replace(findStr, replaceStr)
        file.writeText(updatedContent)
    }
}

tasks.register<GraphQLCodegenFixesGradleTask>("graphQLCodegenFixes") {
    dependsOn("graphqlCodegen")
    inputFile.set(file("$buildDir/generated/com/scentbird/web/graphql/model/MobileLandingGridDataGraph.java"))
}
``` 

**Describe the solution you'd like**
The solution I'm proposing is the following: 
1. Add a new property for a plugin:
`fieldsWithDataFetcherResult = setOf("@dataFetcherResult")`

This property will be used to mark fields that require DataFetcherResult generation. For example:
```java
type MobileLandingGridData {
  # blocks of landing grid
  blocks: [MobileLandingBlock!]! @dataFetcherResult
  # there are more blocks on the next page
  hasNext: Boolean!
}
``` 
2. Update the generator to be able to generate this code.

**Comment by kirill071992** [comment]
@kobylynskyi Hi, any thoughts or suggestions? :)

**Comment by Andras-Csanyi** [comment]
I'm also looking for a solution to being able to generate DataFetcherResult.

**Comment by kobylynskyi** [comment]
Will be included in the upcoming 5.10.0 release

**Comment by kobylynskyi** [comment]
This was released in 5.10.0. Thanks!
